internal/parser/simple_parser_rules.go

```diff
@@ -1928,6 +1930,25 @@ func (p *simpleParser) parseResultColumn(r reporter) (stmt *ast.ResultColumn) {
 				p.consumeToken()
 			}
 		}
+	default:
+		stmt.Expr = p.parseExpression(r)
+		next, ok := p.optionalLookahead(r)
+		if !ok || next.Type() == token.EOF {
+			return
+		}
+		if next.Type() == token.KeywordAs {
+			stmt.As = next
+			p.consumeToken()
+		}
+
+		next, ok = p.optionalLookahead(r)
+		if !ok || next.Type() == token.EOF {
+			return
+		}
+		if next.Type() == token.Literal {
+			stmt.ColumnAlias = next
+			p.consumeToken()
+		}
```

Comment:
code duplication?

Reply:
L1916-1931

Reply:
Yes, multiple cases, internal to it, dont think I can refactor 
.

Reply:
Why not?

Reply:
I think it might lead to some false positives if I add it at the end. Let me see

Reply:
The `AS` and `ColumnAlias` occur only when expr occurs so I cant add it outside the switch case and it must occur both cases of occurrences of `expr` (the literal and non literal one)

And if I use `fallthrough` in the above "repeated coed segment", the second segment parses `expr` again which is not possible.

I'm not sure if this can be refactored, please lmk if you find a way :D

Reply:
Not right now, I see it now.